src/storage/mod.rs

```diff
@@ -4408,4 +4468,120 @@ mod tests {
             .unwrap();
         rx.recv().unwrap();
     }
+
+    #[test]
+    fn test_check_txn_status() {
+        let storage = TestStorageBuilder::new().build().unwrap();
+        let (tx, rx) = channel();
+
+        let k = Key::from_raw(b"k");
+        let v = b"b".to_vec();
+
+        let ts = mvcc::compose_ts;
+
+        // No lock and no commit info. Gets nothing.
+        storage
+            .async_check_txn_status(
+                Context::default(),
+                k.clone(),
+                ts(10, 0),
+                ts(12, 0),
+                ts(15, 0),
+                expect_value_callback(tx.clone(), 0, (0, 0)),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        let mut options = Options::default();
+        options.lock_ttl = 100;
+        storage
+            .async_prewrite(
+                Context::default(),
+                vec![Mutation::Put((k.clone(), v.clone()))],
+                k.as_encoded().to_vec(),
+                ts(10, 0),
+                options.clone(),
+                expect_ok_callback(tx.clone(), 0),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        // If lock exists and not expired, returns the lock's TTL.
+        storage
+            .async_check_txn_status(
+                Context::default(),
+                k.clone(),
+                ts(10, 0),
+                ts(12, 0),
+                ts(15, 0),
+                expect_value_callback(tx.clone(), 0, (100, 0)),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        // TODO: Check the lock's min_commit_ts field.
```

Comment:
Will change `commit()` in another PR and check it?

Reply:
Yes, and due to the ts collision issue, that should be done after I submit and merge another PR to solve the problem.